app/code/Magento/Inventory/Model/GetAssignedSourcesForStock.php

```diff
@@ -0,0 +1,107 @@
+<?php
+/**
+ * Copyright © Magento, Inc. All rights reserved.
+ * See COPYING.txt for license details.
+ */
+namespace Magento\Inventory\Model;
+
+use Magento\Framework\Api\SearchCriteria\CollectionProcessorInterface;
+use Magento\Framework\Api\SearchCriteriaBuilder;
+use Magento\Framework\Exception\InputException;
+use Magento\Framework\Exception\LocalizedException;
+use Magento\Inventory\Model\ResourceModel\StockSourceLink\Collection;
+use Magento\Inventory\Model\ResourceModel\StockSourceLink\CollectionFactory;
+use Magento\InventoryApi\Api\Data\SourceInterface;
+use Magento\InventoryApi\Api\GetAssignedSourcesForStockInterface;
+use Magento\InventoryApi\Api\SourceRepositoryInterface;
+use Psr\Log\LoggerInterface;
+
+/**
+ * @inheritdoc
+ */
+class GetAssignedSourcesForStock implements GetAssignedSourcesForStockInterface
+{
+    /**
+     * @var CollectionProcessorInterface
+     */
+    private $collectionProcessor;
+
+    /**
+     * @var CollectionFactory
+     */
+    private $stockLinkCollectionFactory;
+
+    /**
+     * @var SearchCriteriaBuilder
+     */
+    private $searchCriteriaBuilder;
+
+    /**
+     * @var SourceRepositoryInterface
+     */
+    private $sourceRepository;
+
+    /**
+     * @var LoggerInterface
+     */
+    private $logger;
+
+    /**
+     * @param CollectionProcessorInterface $collectionProcessor
+     * @param CollectionFactory $stockLinkCollectionFactory
+     * @param SearchCriteriaBuilder $searchCriteriaBuilder
+     * @param SourceRepositoryInterface $sourceRepository
+     * @param LoggerInterface $logger
+     */
+    public function __construct(
+        CollectionProcessorInterface $collectionProcessor,
+        CollectionFactory $stockLinkCollectionFactory,
+        SearchCriteriaBuilder $searchCriteriaBuilder,
+        SourceRepositoryInterface $sourceRepository,
+        LoggerInterface $logger
+    ) {
+        $this->collectionProcessor = $collectionProcessor;
+        $this->stockLinkCollectionFactory = $stockLinkCollectionFactory;
+        $this->searchCriteriaBuilder = $searchCriteriaBuilder;
+        $this->sourceRepository = $sourceRepository;
+        $this->logger = $logger;
+    }
+
+    /**
+     * @inheritdoc
+     */
+    public function execute($stockId)
+    {
+        if (0 === (int)$stockId) {
```

Comment:
Because  ctype_digit works with "strings" only 
┌──────────┬───────────┬────────────────┐
│          │  is_int:  │  ctype_digit:  │
├──────────┼───────────┼────────────────┤
│ 123      │  true     │  false         │
├──────────┼───────────┼────────────────┤
│ 12.3     │  false    │  false         │
├──────────┼───────────┼────────────────┤
│ "123"    │  false    │  true          │
├──────────┼───────────┼────────────────┤
│ "12.3"   │  false    │  false         │
├──────────┼───────────┼────────────────┤
│ "-1"     │  false    │  false         │
├──────────┼───────────┼────────────────┤
│ -1       │  true     │  false         │
└──────────┴───────────┴────────────────┘